modular_skyrat/modules/overweight/code/overweight_quirk.dm

```diff
@@ -0,0 +1,153 @@
+		return
+	if(!has_gravity(get_turf(sitter))) //If the sitter is weightless, then they aren't all that heavy
+		return
+	if((sitter.movement_type & FLOATING)) //Floating also makes a person less heavy
+		return
+	/// Only show this stuff to people who also have the quirk
+	if(HAS_TRAIT(sitter, TRAIT_FAT)) //Double fat? start crushing that chair
+		var/make_noise = prob(OVERWEIGHT_AND_FAT_CHAIR_COMPLAIN_CHANCE)
+		var/sit_damage = max_integrity * OVERWEIGHT_AND_FAT_CHAIR_DAMAGE_MULTIPLIER * (sitter.nutrition / NUTRITION_LEVEL_FAT)
+		if((atom_integrity - sit_damage) <= (max_integrity * integrity_failure)) // If the seat can't survive this ass attack, do some special behavior
+			overweight_chair_destruction(sitter)
+			return
+		else if(just_sat_down)
+			sitter.visible_message(span_notice("[src] bends visibly, starting to buckle under [sitter]'s tremendous weight..."),
+			blind_message = span_notice("[src] creaks ominously under [sitter]'s tremendous weight..."),
+			self_message = span_warning("[src] lets out an ominous creak as it struggles to support your tremendous weight!"),
+			ignored_mobs = get_overweight_in_range(sitter, DEFAULT_MESSAGE_RANGE))
+		else if(make_noise)
+			sitter.visible_message(span_notice("[src] bends as it struggles in vain to support [sitter]..."),
+			blind_message = span_notice("You hear a loud creak."),
+			self_message = span_warning("You feel something in \the [src] snap as it tries in vain to support your mass..."),
+			ignored_mobs = get_overweight_in_range(sitter, DEFAULT_MESSAGE_RANGE))
+		if(take_damage(sit_damage, damage_type = BRUTE, damage_flag = MELEE, sound_effect = FALSE)) //If it still didnt do any damage, don't try again
+			addtimer(CALLBACK(src, .proc/overweight_chair_damage, sitter, FALSE), 5 SECONDS)
+	else
+		to_chat(sitter, span_notice("You subject \the [src] to your weight. It creaks, but holds firm."))
+
+/**
+ * This chair just got destroyed by someone's butt. Throw them on the ground
+ *
+ * Arguments:
+ * * seated - Who is sitting in this chair?
+*/
+/obj/structure/chair/proc/overweight_chair_destruction(mob/living/carbon/human/seated)
+	if(!ishuman(seated))
+		return
+	if(seated.buckled != src)
+		return
+	if(!uses_integrity)
+		return
+	if(resistance_flags & INDESTRUCTIBLE)
+		return
+	if(!HAS_TRAIT(seated, TRAIT_OVERWEIGHT))
+		return
+	unbuckle_mob(seated)
+	seated.AdjustKnockdown(5 SECONDS)
+	seated.visible_message(span_warning("[src] falls apart!"),
+	blind_message = span_warning("You hear a loud crash, followed by a soft whump."),
+	self_message = span_warning("You feel something in \the [src] snap as the last of its strength gives out! It crumbles under your weight, throwing you down onto its flattened remains!"))
+	playsound(src, 'modular_skyrat/modules/oversized/sound/chair_break.ogg', 70, TRUE)
+	deconstruct(FALSE)
+
+/obj/structure/chair/proc/get_overweight_in_range(mob/living/carbon/human/origin, range = 2)
```

Comment:
Doesn't this actually do the opposite of what the name implies? It gets NOT overweight mobs in range. The name is a little misleading